### ISSUE TITLE
Quantize/de-quantize inputs/outputs & scaling loss? 

### ISSUE THREAD

**Issue by meetvadera** [Xilinx/brevitas#5]
Hi,

Is it required to quantize inputs or dequantize outputs before passing it to the loss function? Also, does any kind of scaling needs to be performed for the loss function?  

Meet

**Comment by volcacius** [comment]
Not necessary. brevitas.nn layers always return values in dequantized range.

**Comment by meetvadera** [comment]
Ok. Also, would you advise using the PyTorch's functional version of sigmoid/log_sigmoid, or that of brevitas? My understanding is that PyTorch's functional version of sigmoid operates on FP32 which can give better resolution versus that of brevitas. Hence, it might be better to use PyTorch's sigmoid if we are not too concerned about the cost of that particular sigmoid function.

**Comment by volcacius** [comment]
It really depends on your specific use case. What QuantSigmoid does is take a FP32 input in, apply a sigmoid activation function, and the quantize the output according to your specification.

**Comment by meetvadera** [comment]
Ok. I am particularly talking about the use case of training for classification. In that case, I believe F.sigmoid will be a better choice given that I'd want to use F.nll_loss downstream for training and it'll give me better resolution.

**Comment by volcacius** [comment]
If your accuracy is unreasonably high, it might that you have quantization disabled. The default behavior is to have quantization disabled, i.e. a QuantConv2d behaves by default as a Conv2d.. To enable integer quantization you need to set weight_quant_type = QuantType.INT . I understand this might be confusing, so i'll force the user to specify the QuantType in a later update.

Regarding dequantization, it is not exposed to the user as a separate layer. It's performed [here](https://github.com/Xilinx/brevitas/blob/3988cf4a5e6b1d076c4f84911b3f3d3732abe233/brevitas/core/quant.py#L289) and is called as part of every quantized layer. 

What those flags do is to compute explicitly the scale factor of the output accumulator, as well as its maximum bit width, and return them as quantized tensor, which is a named tuple composed of (output_tensor, output_scale_factor, output_bit_width). The relationship between those values is that output_tensor/output_scale_factors are integer values (when you have bias disabled) that can be represented with output_bit_width bits. Enabling them is not required in general. 

This sort of information is gonna go in the documentation as soon the API stabilize enough.
